diff --git a/mono.SlackBuild b/home/justin/Projects/slackbuilds/mono/mono.SlackBuild
old mode 100644
new mode 100755
index 00681d3d7e..141f8b57c7
--- a/mono.SlackBuild
+++ b/home/justin/Projects/slackbuilds/mono/mono.SlackBuild
@@ -5,9 +5,7 @@
 # Updated by Andre Barboza <bmg.andre@gmail.com>
 
 PRGNAM=mono
-VERSION=${VERSION:-5.0.1.1}
-MAJORVERSION=${MAJORVERSION:-5.0.1.1}
-BIN_VERSION=${BIN_VERSION:-4.4.2.11}
+VERSION=${VERSION:-5.20.1.34}
 BUILD=${BUILD:-1}
 TAG=${TAG:-_SBo}
 
@@ -49,8 +47,7 @@ mkdir -p $TMP $PKG $OUTPUT
 cd $TMP
 rm -rf $PRGNAM-$VERSION
 tar xvf $CWD/$PRGNAM-$VERSION.tar.bz2
-cd $PRGNAM-$MAJORVERSION
-
+cd $PRGNAM-$VERSION
 chown -R root:root .
 find -L . \
  \( -perm 777 -o -perm 775 -o -perm 750 -o -perm 711 -o -perm 555 \
@@ -58,13 +55,6 @@ find -L . \
  \( -perm 666 -o -perm 664 -o -perm 640 -o -perm 600 -o -perm 444 \
   -o -perm 440 -o -perm 400 \) -exec chmod 644 {} \;
 
-# Use full featured precompiled mono version instead of monolite
-pushd $TMP
-rm -rf $PRGNAM-$BIN_VERSION-$BOOTSTRAP_ARCH
-tar xvf $CWD/$PRGNAM-$BIN_VERSION-$BOOTSTRAP_ARCH.tar.bz2
-export PATH="$TMP/$PRGNAM-$BIN_VERSION-$BOOTSTRAP_ARCH/usr/bin:$PATH"
-popd
-
 CFLAGS="$SLKCFLAGS" \
 CXXFLAGS="$SLKCFLAGS" \
 ./configure \
